### PR TITLE
feat: Add Answers to Questions

### LINKED ISSUE #14: We should create an answer model with a one to many relationship with Question
We should create an answer model with a one to many relationship with Question

_Originally posted by @bens-schreiber in https://github.com/bens-schreiber/CPTS-322-Course-Project/issues/13#issuecomment-2076104083_

### PR DESCRIPTION
**Pull request by DamiennFl**
fixes #14 

Models are for sure correct now.

Session is just a session, poll is connected to a session with one to one, question is connected to a poll with one to one, answer is connected to question with many to one, and response is connected to poll with a many to one, and each response has a one to one with an answer

### CODE REVIEW

File: .github/workflows/django.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
+name: Django Run Tests
```
Comment (bens-schreiber) [inline .github/workflows/django.yml:1]: this file should be deleted, bad merge

File: .github/workflows/formatter.yml [hunk 2]
```diff
@@ -0,0 +1,31 @@
+name: Django Run Formatter
```
Comment (bens-schreiber) [inline .github/workflows/formatter.yml:1]: this file should be deleted, bad merge

File: backend/pollpall_project/session/models.py [hunk 3]
```diff
@@ -2,72 +2,82 @@
 
 
 class Session(models.Model):
```
Comment (bens-schreiber) [inline backend/pollpall_project/session/models.py:4]: Session needs a poll.

File: backend/pollpall_project/session/models.py [hunk 4]
```diff
@@ -2,72 +2,82 @@
...
+    is_accepting_answers (bool): Whether the poll is accepting answers or not
     """
 
-    poll_id = models.IntegerField(primary_key=True)
-    session = models.ForeignKey("Session", on_delete=models.CASCADE)
-    question_id = models.IntegerField(default=-1)
-    related_question = models.ForeignKey(
-        "Question", on_delete=models.CASCADE, related_name="related_polls"
-    )
+    session = models.OneToOneField("Session", on_delete=models.CASCADE)
```
Comment (bens-schreiber) [inline backend/pollpall_project/session/models.py:28]: Poll does not need a session
```suggestion
```

File: backend/pollpall_project/session/models.py [hunk 5]
```diff
@@ -2,72 +2,82 @@
...
+    question = models.OneToOneField("Question", on_delete=models.CASCADE)
     is_accepting_answers = models.BooleanField(default=False)
 
     def __str__(self):
-        return f"{self.poll_id}: {self.session}: {self.question_id}: {self.related_question}: {self.is_accepting_answers}"
+        return f"{self.is_accepting_answers}"
 
 
 class Question(models.Model):
```
Comment (bens-schreiber) [inline backend/pollpall_project/session/models.py:36]: Question needs an answer.

File: backend/pollpall_project/session/models.py [hunk 6]
```diff
@@ -2,72 +2,82 @@
...
+    Attributes
+    answer (str): Text of the answer
+    question (Question): The question this answer is connected to
+    is_correct (bool): Whether this answer is correct or not
     """
 
-    response_id = models.IntegerField(primary_key=True)
-    prompt = models.CharField(max_length=0xFFF, default="")
+    answer = models.CharField(max_length=0xFFF, default="")
     question = models.ForeignKey("Question", on_delete=models.CASCADE)
```
Comment (bens-schreiber) [inline backend/pollpall_project/session/models.py:61]: Answers do not need a question

File: backend/pollpall_project/session/models.py [hunk 7]
```diff
@@ -2,72 +2,82 @@
...
+    Response Model
+
+    Attributes
+    answer (Answer): The answer which was chosen
+    poll (Poll): The poll this response is connected to
+    is_correct (bool): Whether the response was correct or not
+    """
+
+    answer = models.OneToOneField("Answer", on_delete=models.CASCADE)
+    poll = models.ForeignKey("Poll", on_delete=models.CASCADE)
```
Comment (bens-schreiber) [inline backend/pollpall_project/session/models.py:79]: Responses do not need a poll

File: backend/pollpall_project/session/tests.py [hunk 8]
```diff
@@ -1,7 +1,7 @@
 from django.urls import reverse
 from django.test import Client, TestCase
 
-from .models import Session, Question
+from .models import Poll, Session, Question
 from .views import Session
 
 SESSION_POST = reverse("pollpal:session-list-create")
```
Comment (bens-schreiber) [inline backend/pollpall_project/session/tests.py:7]: Delete session tests

**Review by bens-schreiber — CHANGES_REQUESTED** [review]

File: backend/pollpall_project/session/models.py [hunk 9]
```diff
@@ -2,72 +2,82 @@
...
     """
 
-    poll_id = models.IntegerField(primary_key=True)
-    session = models.ForeignKey("Session", on_delete=models.CASCADE)
-    question_id = models.IntegerField(default=-1)
-    related_question = models.ForeignKey(
-        "Question", on_delete=models.CASCADE, related_name="related_polls"
-    )
+    session = models.OneToOneField("Session", on_delete=models.CASCADE)
+    question = models.OneToOneField("Question", on_delete=models.CASCADE)
```
Comment (bens-schreiber) [inline backend/pollpall_project/session/models.py:29]: Poll shouldn't have a question, a question should have a OneToOne with a Poll which will give the Poll obj a question

File: backend/pollpall_project/session/models.py [hunk 10]
```diff
@@ -2,72 +2,82 @@
...
+    """
+    Response Model
+
+    Attributes
+    answer (Answer): The answer which was chosen
+    poll (Poll): The poll this response is connected to
+    is_correct (bool): Whether the response was correct or not
+    """
+
+    answer = models.OneToOneField("Answer", on_delete=models.CASCADE)
```
Comment (bens-schreiber) [inline backend/pollpall_project/session/models.py:78]: Answer should have a OneToOne with response

File: backend/pollpall_project/pollpall_project/settings.py [hunk 11]
```diff
@@ -31,7 +31,6 @@
 # Application definition
 
 INSTALLED_APPS = [
-    "drf_spectacular",
```
Comment (bens-schreiber) [inline backend/pollpall_project/pollpall_project/settings.py:34]: why did you remove this

File: backend/pollpall_project/session/views.py [hunk 12]
```diff
@@ -1,3 +1,4 @@
+from django.http import JsonResponse
```
Comment (bens-schreiber) [inline backend/pollpall_project/session/views.py:1]: delete this and delete lookup_field from the views in this file

File: backend/pollpall_project/session/models.py [hunk 13]
```diff
@@ -2,72 +2,82 @@
...
+
+    Attributes
+    answer (str): Text of the answer
+    question (Question): The question this answer is connected to
+    is_correct (bool): Whether this answer is correct or not
+    """
+
+    answer = models.CharField(max_length=0xFFF, default="")
+    response = models.OneToOneField("Response", on_delete=models.CASCADE)
+    question = models.ForeignKey("Question", on_delete=models.CASCADE, null=True)
```
Comment (bens-schreiber) [inline backend/pollpall_project/session/models.py:62]: I think a Question should be required to have an answer, null is bad

**Review by bens-schreiber — CHANGES_REQUESTED** [review]

File: backend/pollpall_project/pollpall_project/settings.py (hunk 11 above)
Comment (DamiennFl) [reply backend/pollpall_project/pollpall_project/settings.py:34]: idk it works on mine without it, i'll add it back

File: backend/pollpall_project/session/models.py (hunk 13 above)
Comment (DamiennFl) [reply backend/pollpall_project/session/models.py:62]: I only did that so when we test the services, we don't have to make a bunch of objects for testing, that works tho

File: backend/pollpall_project/pollpall_project/settings.py (hunk 11 above)
Comment (bens-schreiber) [reply backend/pollpall_project/pollpall_project/settings.py:34]: its for Swagger

Comment (DamiennFl) [reply backend/pollpall_project/pollpall_project/settings.py:34]: ok

**Review by bens-schreiber — APPROVED** [review]
